File: packages/ui_footer/tests/metadata.test.ts
import { describe, it, expect } from 'vitest'
import metadata from '../seed/metadata.json'

describe('UI Footer Metadata', () => {
  it('should have valid structure', () => {
    expect(metadata.packageId).toBe('ui_footer')
    expect(metadata.version).toMatch(/^\d+\.\d+\.\d+$/)
  })
  it('should export AppFooter component', () => {
    expect(metadata.exports.components).toContain('AppFooter')
  })
})

File: packages/archive/ui_footer/seed/scripts/tests/metadata.test.lua
-- Metadata validation tests for ui_footer package
-- Uses lua_test framework

describe("App Footer Package Metadata", function()
  local metadata = load_cases("metadata.json")
  
  it("should have valid package structure", function()
    expect(metadata.packageId).toBe("ui_footer")
    expect(metadata.name).toBe("App Footer")
    expect(metadata.version).toBeTruthy()
    expect(metadata.description).toBeTruthy()
  end)
  
  it("should have correct package ID format", function()
    expect(metadata.packageId).toMatch("^[a-z_]+$")
  end)
  
  it("should have semantic version", function()
    expect(metadata.version).toMatch("^%d+%.%d+%.%d+$")
  end)
  
  it("should have exports defined", function()
    expect(metadata.exports).toBeTruthy()
    expect(metadata.exports.components).toBeType("table")
  end)
  
  it("should have dependencies array", function()
    expect(metadata.dependencies).toBeType("table")
  end)
end)
